component/src/main/java/io/siddhi/extension/store/elasticsearch/sink/ElasticsearchSink.java

```diff
@@ -0,0 +1,287 @@
+                    elasticsearchConfigs.getBulkProcessor().add(siddhiIndexRequest);
+                }
+                logger.debug(payload + " has been successfully added.");
+            } else if (parse.isJsonObject()) {
+                siddhiIndexRequest.source(parse.getAsJsonObject().toString(), XContentType.JSON);
+                elasticsearchConfigs.getBulkProcessor().add(siddhiIndexRequest);
+            }
+        } catch (JsonSyntaxException e) {
+            throw new ElasticsearchEventSinkException("Invalid json document, Please recheck the json mapping at" +
+                    "the '@payload'.", e);
+        }
+    }
+
+    @Override
+    public void connect() throws ConnectionUnavailableException {
+        if (elasticsearchConfigs.getIndexName() != null && !elasticsearchConfigs.getIndexName().isEmpty()) {
+            createIndex(elasticsearchConfigs.getDefinitionId());
+        }
+    }
+
+    @Override
+    public void disconnect() {
+        try {
+            elasticsearchConfigs.getRestHighLevelClient().close();
+        } catch (IOException e) {
+            throw new ElasticsearchEventSinkException("Error while closing the connection.", e);
+        } finally {
+            elasticsearchConfigs.setRestHighLevelClient(null);
+        }
+    }
+
+    @Override
+    public void destroy() {
+    }
+
+    private void createIndex(String definitionId) throws ConnectionUnavailableException { //call at the connect
+        try {
+            if (elasticsearchConfigs.getRestHighLevelClient().indices().exists(
+                    new GetIndexRequest(elasticsearchConfigs.getIndexName()), RequestOptions.DEFAULT)) {
+                logger.debug("Index: " + elasticsearchConfigs.getIndexName() + " has already being created for sink " +
+                        "id: " + definitionId + ".");
+                return;
+            }
+        } catch (IOException e) {
+            throw new ConnectionUnavailableException("Error while checking indices for sink id : '" +
+                    definitionId, e);
+        }
+        CreateIndexRequest request = new CreateIndexRequest(elasticsearchConfigs.getIndexName());
+        request.settings(Settings.builder()
+                        .put(SETTING_INDEX_NUMBER_OF_SHARDS, elasticsearchConfigs.getNumberOfShards())
+                        .put(SETTING_INDEX_NUMBER_OF_REPLICAS, elasticsearchConfigs.getNumberOfReplicas()));
+        if (elasticsearchConfigs.getIndexAlias() != null) {
+            request.alias(new Alias(elasticsearchConfigs.getIndexAlias()));
+        }
+        try {
+            elasticsearchConfigs.getRestHighLevelClient().indices().create(request, RequestOptions.DEFAULT);
+            logger.debug("A sink id: " + definitionId + " is created with the provided information.");
+        } catch (IOException e) {
+            throw new ElasticsearchEventSinkException("Error while creating indices for sink id : '" +
+                    definitionId, e);
```

Comment:
Isn't using definitionID for sinkID wrong ?

Reply:
Check below log as well.

Reply:
Fixed in https://github.com/siddhi-io/siddhi-store-elasticsearch/pull/41/commits/8887cc35f24420f2482087ac0c3b1e1aea2e6ccf commit